wrapper/minecraft/src/main/kotlin/org/quiltmc/qkl/wrapper/minecraft/text/TextStyles.kt

```diff
@@ -0,0 +1,180 @@
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+@file:Suppress("MemberVisibilityCanBePrivate")
+
+package org.quiltmc.qkl.wrapper.minecraft.text
+
+import net.minecraft.text.*
+import net.minecraft.util.Formatting
+import org.quiltmc.qkl.core.annotations.QklDslMarker
+
+/**
+ * A mutable color object that can be transformed into Minecraft [color][TextColor].
+ *
+ * @property rgb The RGB value to convert to a Minecraft [color][TextColor].
+ *
+ * @author NoComment1105
+ */
+@QklDslMarker
+public class MutableColor(public var rgb: Int) {
+    public companion object {
+        /**
+         * Gets a color from [TextColor] and converts it to [MutableColor].
+         *
+         * @param color The [TextColor] to convert
+         * @return A [MutableColor] from [color]
+         *
+         * @author NoComment1105
+         */
+        public fun fromTextColor(color: TextColor): MutableColor {
+            return MutableColor(color.rgb)
+        }
+
+        /**
+         * Gets a color from [Formatting] and converts it to [MutableColor].
+         * If the color value is null, it defaults to black.
+         *
+         * @param formattingColor The color to convert
+         * @return A [MutableColor] from [formattingColor] or black if invalid/null
+         *
+         * @author NoComment1105
+         */
+        public fun fromFormatting(formattingColor: Formatting): MutableColor {
+            return MutableColor(formattingColor.colorValue ?: 0x000000)
+        }
+    }
+
+    /**
+     * Converts [rgb] to a [TextColor]
+     *
+     * @return A [TextColor] created from the [rgb] value
+     *
+     * @author NoComment1105
+     */
+    public fun toTextColor(): TextColor {
+        return TextColor.fromRgb(rgb)
+    }
+}
+
+
+/**
+ * A mutable style object that transforms to Minecraft [styles][Style].
+ *
+ * @author NoComment1105
+ */
+@QklDslMarker
+@Suppress("MagicNumber")
+public class MutableStyle {
+    /** A [MutableColor] to apply to the text. */
+    public var color: MutableColor? = null
+    /** Whether to obfuscate the text or not. */
+    public var obfuscated: Boolean = false
+    /** Whether to format the text in bold or not. */
+    public var bold: Boolean = false
+    /** Whether to format the text in italics or not. */
+    public var italic: Boolean = false
+    /** Whether to format the text with a strikethrough or not. */
+    public var strikethrough: Boolean = false
+    /** Whether to format the text with an underline or not. */
+    public var underline: Boolean = false
+    /** A [HoverEvent] to apply to the text. */
+    public var hoverEvent: HoverEvent? = null
```

Comment:
Needs DSL functions imo. [Reference](https://github.com/SpaceServe/spaceserve-ekho/wiki/Hover-and-Click-Events)